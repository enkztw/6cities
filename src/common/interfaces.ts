import {city, id, property} from './types';

export interface Location {
    latitude: number;
    longitude: number;
    zoom: number;
}

export interface City {
    location: Location;
    name: city;
}

export interface User {
    readonly id: id;
    avatarUrl: string;
    isPro: boolean;
    name: string;
    email?: string;
    description?: string;
}

export interface Review {
    readonly id: id;
    comment: string;
    date: string;
    rating: number;
    user: User;
}

export interface Status {
    isAuthorized: boolean;
}

export interface Offer {
    readonly id: id;
    bedrooms: number;
    city: City;
    description: string;
    goods: string[];
    host: User;
    images: string[];
    isFavorite: boolean;
    isPremium: boolean;
    location: Location;
    maxAdults: number;
    previewImage: string;
    price: number;
    rating: number;
    title: string;
    type: property;
}
