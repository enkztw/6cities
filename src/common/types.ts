export type property = `apartment` | `room` | `house` | `hotel`;
export type id = number
export type city = `Paris` | `Cologne` | `Brussels` | `Amsterdam` | `Hamburg` | `Dusseldorf`
export type sort = `rating` | `id` | `toLow` | `toHigh`


