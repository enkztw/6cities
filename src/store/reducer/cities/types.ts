enum Types {
  SET_CITY = `SET_CITY`
}

export default Types;
