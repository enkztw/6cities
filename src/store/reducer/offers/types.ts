enum Types {
  SET_OFFER = `SET_OFFER`,
  SET_COMMENTS = `SET_COMMENTS`,
  SET_OFFERS = `SET_OFFERS`,
  SET_FAVORITES = `SET_FAVORITES`,
  SET_NEARBY = `SET_NEARBY`,
  SET_SORTING = `SET_SORTING`
}

export default Types;
