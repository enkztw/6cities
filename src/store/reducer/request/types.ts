enum Types {
  SET_REQUEST = `SET_REQUEST`,
}

export default Types;

