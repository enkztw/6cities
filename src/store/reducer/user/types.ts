enum Types {
  SET_AUTH = `SET_AUTH`
}

export default Types;

