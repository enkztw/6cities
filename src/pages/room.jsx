import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {ActionCreator} from '../store/actions/actions.js';

import Header from '../components/header/header.jsx';
import Gallery from '../components/gallery/gallery.jsx';
import Intro from '../components/offer-intro/offer-intro.jsx';
import Features from '../components/features/features.jsx';
import Host from '../components/host/host.jsx';
import Reviews from '../components/reviews/reviews.jsx';
import Map from '../components/map/map.jsx';
import Offers from '../components/offers/offers-nearby.jsx';

class Room extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.init(this.props.offers, parseInt(this.props.match.params.id, 10));
    this.props.getComments(parseInt(this.props.match.params.id, 10));
  }

  componentDidUpdate() {
    if ((this.props.currentOffer) && (this.props.match.params.id !== this.props.currentOffer.id)) {
      window.scrollTo({
        top: 0,
        behavior: `smooth`
      });

      this.props.init(this.props.offers, parseInt(this.props.match.params.id, 10));
      this.props.getComments(parseInt(this.props.match.params.id, 10));
    }
  }

  render() {
    const {currentOffer, offers, comments} = this.props;

    if (!currentOffer) {
      return null;
    }

    const {id, images, goods, host, city} = currentOffer;
    const nearbyOffers = offers.filter((offer) => (offer.city.name === city.name) && (offer.id !== id));

    return (
      <div className="page">
        <Header />
        <main className="page__main page__main--property">
          <section className="property">
            <Gallery images={images} />
            <div className="property__container container">
              <div className="property__wrapper">
                <Intro {...currentOffer} />
                <Features features={goods}/>
                <Host {...host}/>
                <Reviews reviews={comments}/>
              </div>
            </div>
            <Map offers={nearbyOffers} currentOffer={currentOffer}/>
          </section>
          {nearbyOffers.length > 0 && <Offers offers={nearbyOffers}/>}
        </main>
      </div>
    );
  }
}

Room.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string
    })
  }),
  offers: PropTypes.array,
  currentOffer: PropTypes.shape({
    id: PropTypes.string,
    images: PropTypes.array,
    goods: PropTypes.array,
    host: PropTypes.object,
    reviews: PropTypes.array,
    city: PropTypes.shape({
      name: PropTypes.string
    })
  }),
  comments: PropTypes.array,
  nearbyOffers: PropTypes.array,
  init: PropTypes.func,
  getComments: PropTypes.func
};

const mapStateToProps = (state) => ({
  offers: state.offers,
  currentOffer: state.currentOffer,
  comments: state.comments
});


const mapDispatchToProps = (dispatch) => ({
  init(offers, id) {
    dispatch(ActionCreator.initOffer(offers, id));
  },
  getComments(id) {
    dispatch(ActionCreator.getComments(id));
  }
});
export default connect(mapStateToProps, mapDispatchToProps)(Room);
