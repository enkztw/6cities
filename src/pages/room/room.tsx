import * as React from 'react';
import {connect} from 'react-redux';

import {Offer} from '../../common/interfaces';
import {getCity} from '../../store/reducer/cities/selectors';
import {getOffer, getNearbyOffers as getOffers} from '../../store/reducer/offers/selectors';
import {ActionCreator as OffersActionCreator, getNearby, setFavorite, getComments} from '../../store/reducer/offers/actions';
import {ActionCreator as CitiesActionCreator} from '../../store/reducer/cities/actions';

import Header from '../../components/header/header';
import Gallery from '../../components/gallery/gallery';
import Intro from '../../components/offer-intro/offer-intro';
import Features from '../../components/features/features';
import Host from '../../components/host/host';
import Reviews from '../../components/reviews/reviews';
import Map from '../../components/map/map';
import Offers from '../../components/offers/offers-nearby';
import {MAX_IMAGES, OperationStatus} from '../../common/const';


interface Props {
  match: {
    params: {
      id: string;
    };
  };
  offer: Offer;
  offers: Offer[];
  city: string;
  status: string;
  setOffer: (id: number) => void;
  getComments: (id: string) => void;
  getNearby: (id: string) => void;
  setCity: (name: string) => void;
  setFavorite: (id: number, status: number) => void;
}

class Room extends React.PureComponent<Props, {}> {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  componentDidMount() {
    const id = this.props.match.params.id;
    this.props.setOffer(parseInt(id, 10));
    this.props.getComments(id);
    this.props.getNearby(id);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.match.params.id !== this.props.match.params.id) {
      window.scrollTo({
        top: 0,
        behavior: `smooth`
      });

      const id = this.props.match.params.id;
      this.props.setOffer(parseInt(id, 10));
      this.props.getComments(id);
      this.props.getNearby(id);
    }

    if (this.props.offer && (this.props.city !== this.props.offer.city.name)) {
      this.props.setCity(this.props.offer.city.name);
    }
  }

  handleClick() {
    const {offer: {id, isFavorite}} = this.props;
    const status = isFavorite ? 0 : 1;
    this.props.setFavorite(id, status);
  }

  render() {
    const {offer, offers, status} = this.props;

    if (!offer) {
      return null;
    }

    const {images, goods, host, description} = offer;
    return (
      <div className="page">
        <Header />
        <main className="page__main page__main--property">
          <section className="property">
            <Gallery images={images.slice(0, MAX_IMAGES)} />
            <div className="property__container container">
              <div className="property__wrapper">
                <Intro {...offer} onClick={this.handleClick} errorStatus={status === OperationStatus.FAILED}/>
                <Features goods={goods}/>
                <Host {...host} description={description}/>
                <Reviews />
              </div>
            </div>
            <Map offers={[...offers, offer]} currentOffer={offer.id}/>
          </section>
          {offers.length > 0 && <Offers offers={offers}/>}
        </main>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  city: getCity(state),
  offers: getOffers(state),
  offer: getOffer(state),
  status: state.request.type === `favorite` ? state.request.status : ``
});


const mapDispatchToProps = (dispatch) => ({
  setCity(name) {
    dispatch(CitiesActionCreator.setCity(name));
  },
  setOffer(id) {
    dispatch(OffersActionCreator.setOffer(id));
  },
  getComments(id) {
    dispatch(getComments(id));
  },
  getNearby(id) {
    dispatch(getNearby(id));
  },
  setFavorite(id, status) {
    dispatch(setFavorite(id, status));
  }
});

export {Room};
export default connect(mapStateToProps, mapDispatchToProps)(Room);
