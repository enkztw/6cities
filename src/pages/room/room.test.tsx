import * as React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';
import configureStore from 'redux-mock-store';

import {Offer, Review} from '../../interfaces';
import {city} from '../../types';
import {Room} from './room';


const mockStore = configureStore([]);

const cities: city[] = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
const offers: Offer[] = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 1.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 100,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  }
];

const nearby: Offer[] = [
  {
    id: 2,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    previewImage: `img/apartment-02.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 2.8,
    type: `room`,
    bedrooms: 2,
    maxAdults: 4,
    price: 140,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8
    }
  },
  {
    id: 3,
    city: {
      name: `Paris`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    previewImage: `img/apartment-02.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `room`,
    bedrooms: 2,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8
    }
  }
];

const comments: Review[] = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatarUrl: `/img/avatar-max.jpg`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];

it(`<Room /> should be rendered`, () => {
  const div = document.createElement(`div`);
  document.body.appendChild(div);

  const store = mockStore({
    user: {
      isAuthorized: true,
      email: `jztenk@gmail.com`
    },
    cities: {
      data: cities,
      current: `Amsterdam`
    },
    offers: {
      sorting: `id`,
      data: offers,
      nearby,
      current: null,
      comments
    },
    request: {
      type: ``,
      status: ``,
      id: null
    }
  });

  const tree = mount(
      <Provider store={store}>
        <Router>
          <Room
            match={{params: {id: `1`}}}
            city={`Amsterdam`}
            offers={offers}
            offer={offers[0]}
            status={``}
            setOffer={jest.fn()}
            setCity={jest.fn()}
            setBookmark={jest.fn()}
            getComments={() => comments}
            getNearbyOffers={() => nearby}
          />
        </Router>
      </Provider>, {attachTo: div});

  expect(tree.getDOMNode()).toMatchSnapshot();
});
