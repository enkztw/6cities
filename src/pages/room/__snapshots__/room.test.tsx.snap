// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Room /> should be rendered 1`] = `
<div
  class="page"
>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link header__logo-link--active"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/favorites"
              >
                <div
                  class="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  class="header__user-name user__name"
                >
                  jztenk@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    class="page__main page__main--property"
  >
    <section
      class="property"
    >
      <div
        class="property__gallery-container container"
      >
        <div
          class="property__gallery"
        >
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            class="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              class="property__image"
              src="img/apartment-03.jpg"
            />
          </div>
        </div>
      </div>
      <div
        class="property__container container"
      >
        <div
          class="property__wrapper"
        >
          <div
            class="property__name-wrapper"
          >
            <h1
              class="property__name"
            >
              Beautiful & luxurious studio at great location
            </h1>
            <button
              class="property__bookmark-button button"
              type="button"
            >
              <svg
                class="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="property__rating rating"
          >
            <div
              class="property__stars rating__stars"
            >
              <span
                style="width: 40%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              class="property__rating-value rating__value"
            >
              1.8
            </span>
          </div>
          <ul
            class="property__features"
          >
            <li
              class="property__feature property__feature--entire"
            >
              apartment
            </li>
            <li
              class="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              class="property__feature property__feature--adults"
            >
              Max 
              4
               adults
            </li>
          </ul>
          <div
            class="property__price"
          >
            <b
              class="property__price-value"
            >
              €
              100
            </b>
            <span
              class="property__price-text"
            >
               night
            </span>
          </div>
          <div
            class="property__inside"
          >
            <h2
              class="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              class="property__inside-list"
            >
              <li
                class="property__inside-item"
              >
                Heating
              </li>
              <li
                class="property__inside-item"
              >
                Kitchen
              </li>
              <li
                class="property__inside-item"
              >
                Cable TV
              </li>
              <li
                class="property__inside-item"
              >
                Washing machine
              </li>
              <li
                class="property__inside-item"
              >
                Coffee machine
              </li>
              <li
                class="property__inside-item"
              >
                Dishwasher
              </li>
            </ul>
          </div>
          <div
            class="property__host"
          >
            <h2
              class="property__host-title"
            >
              Meet the host
            </h2>
            <div
              class="property__host-user user"
            >
              <div
                class="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
              >
                <img
                  alt="Host avatar"
                  class="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                class="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              class="property__description"
            >
              <p>
                A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
              </p>
            </div>
          </div>
          <section
            class="property__reviews reviews"
          >
            <h2
              class="reviews__title"
            >
              Reviews · 
              <span
                class="reviews__amount"
              >
                1
              </span>
            </h2>
            <ul
              class="reviews__list"
            >
              <li
                class="reviews__item"
              >
                <div
                  class="reviews__user user"
                >
                  <div
                    class="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Max"
                      class="reviews__avatar user__avatar"
                      height="54"
                      src="/img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    class="reviews__user-name"
                  >
                    Max
                  </span>
                </div>
                <div
                  class="reviews__info"
                >
                  <div
                    class="reviews__rating rating"
                  >
                    <div
                      class="reviews__stars rating__stars"
                    >
                      <span
                        style="width: 80%;"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    class="reviews__text"
                  >
                    A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  </p>
                  <time
                    class="reviews__time"
                    datetime="2019-05-08T14:13:56.569Z"
                  >
                    May
                     
                    2019
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              class="reviews__form form"
              method="post"
            >
              <label
                class="reviews__label form__label"
                for="review"
              >
                Your review
              </label>
              <div
                class="reviews__rating-form form__rating"
              >
                <input
                  class="form__rating-input visually-hidden"
                  id="5-stars"
                  name="rating"
                  type="radio"
                  value="5"
                />
                <label
                  class="reviews__rating-label form__rating-label"
                  for="5-stars"
                  title="perfect"
                >
                  <svg
                    class="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlink:href="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  class="form__rating-input visually-hidden"
                  id="4-stars"
                  name="rating"
                  type="radio"
                  value="4"
                />
                <label
                  class="reviews__rating-label form__rating-label"
                  for="4-stars"
                  title="perfect"
                >
                  <svg
                    class="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlink:href="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  class="form__rating-input visually-hidden"
                  id="3-stars"
                  name="rating"
                  type="radio"
                  value="3"
                />
                <label
                  class="reviews__rating-label form__rating-label"
                  for="3-stars"
                  title="perfect"
                >
                  <svg
                    class="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlink:href="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  class="form__rating-input visually-hidden"
                  id="2-stars"
                  name="rating"
                  type="radio"
                  value="2"
                />
                <label
                  class="reviews__rating-label form__rating-label"
                  for="2-stars"
                  title="perfect"
                >
                  <svg
                    class="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlink:href="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  class="form__rating-input visually-hidden"
                  id="1-stars"
                  name="rating"
                  type="radio"
                  value="1"
                />
                <label
                  class="reviews__rating-label form__rating-label"
                  for="1-stars"
                  title="perfect"
                >
                  <svg
                    class="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlink:href="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                class="reviews__textarea form__textarea"
                id="review"
                name="review"
                placeholder="Tell how was your stay, what you like and what can be improved"
              />
              <div
                class="reviews__button-wrapper"
              >
                <p
                  class="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    class="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    class="reviews__text-amount"
                  >
                    50
                     and below 
                    300
                     characters
                  </b>
                  .
                </p>
                <button
                  class="reviews__submit form__submit button"
                  disabled=""
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        class="property__map map leaflet-container leaflet-grab leaflet-touch-drag"
        id="map"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="left: 0px; top: 0px;"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; left: 0px; top: 0px;"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  role="presentation"
                  src="https://a.basemaps.cartocdn.com/rastertiles/voyager_labels_under/10/525/336{@2x}.png"
                  style="width: 256px; height: 256px; left: -237px; top: -134px;"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          >
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="img/pin-active.svg"
              style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -162px; top: 18px; z-index: 18;"
              tabindex="0"
            />
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="img/pin-active.svg"
              style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -162px; top: 18px; z-index: 18;"
              tabindex="0"
            />
          </div>
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          />
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          >
            <div
              class="leaflet-control-attribution leaflet-control"
            >
              <a
                href="https://leafletjs.com"
                title="A JS library for interactive maps"
              >
                Leaflet
              </a>
            </div>
          </div>
        </div>
      </section>
    </section>
    <div
      class="container"
    >
      <section
        class="near-places places"
      >
        <h2
          class="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          class="near-places__list places__list"
        >
          <article
            class="cities__place-card place-card"
            data-id="1"
          >
            <div
              class="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  class="place-card__image"
                  height="200"
                  src="img/apartment-01.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              class="place-card__info"
            >
              <div
                class="place-card__price-wrapper"
              >
                <div
                  class="place-card__price"
                >
                  <b
                    class="place-card__price-value"
                  >
                    €
                    100
                  </b>
                  <span
                    class="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  class="place-card__bookmark-button button"
                  type="button"
                >
                  <svg
                    class="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlink:href="#icon-bookmark"
                    />
                  </svg>
                  <span
                    class="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                class="place-card__rating rating"
              >
                <div
                  class="place-card__stars rating__stars"
                >
                  <span
                    style="width: 40%;"
                  />
                  <span
                    class="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                class="place-card__name"
              >
                <a
                  href="/offer/1"
                >
                  Beautiful & luxurious studio at great location
                </a>
              </h2>
              <p
                class="place-card__type"
              >
                apartment
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
